### PR TITLE
fix: mimetypes

### PR DESCRIPTION
**Pull request by tsi**
This PR fixes our mimetypes testing logic.
Until now we were simply using the file extension for non ABR, which was ok for some types but bad for others.
This is now aligned with [videojs mimetypes](https://docs.videojs.com/utils_mimetypes.js.html)

### CODE REVIEW

File: src/plugins/cloudinary/models/video-source/video-source.js [hunk 1]
```diff
@@ -244,10 +244,12 @@ class VideoSource extends BaseSource {
   }
 
   generateRawSource(url, type) {
-    const t = type || url.split('.').pop();
-    const isAdaptive = !!CONTAINER_MIME_TYPES[t];
-    if (isAdaptive) {
-      type = CONTAINER_MIME_TYPES[t][0];
+    type = type || url.split('.').pop();
+
+    const isAdaptive = ['hls', 'dash', 'mpd', 'm3u8'].indexOf(type) !== -1;
```
Comment (jakub-roch) [inline src/plugins/cloudinary/models/video-source/video-source.js:249]: what do you say about moving this list to video source const?

`export const ADAPTIVE_SOURCES = ['hls', 'dash', 'mpd', 'm3u8'];`

File: src/utils/cloudinary.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ const setError = (that, res) => {
 };
 
 const setVideoSrc = (that, srcs) => {
-  console.log('Trying urls: ' + JSON.stringify(srcs));
+  console.log('Trying sources: ', srcs);
```
Comment (jakub-roch) [inline src/utils/cloudinary.js:37]: I see it was previously there but can we get rid of it or hide it behind some debug mode?

**Review by jakub-roch — APPROVED** [review]
Looks good but please see my comments

File: src/plugins/cloudinary/models/video-source/video-source.utils.js [hunk 3]
```diff
@@ -6,16 +6,12 @@ import { some } from '../../../../utils/array';
...
 export function formatToMimeTypeAndTransformation(format) {
   const [container, codec] = format.toLowerCase().split('\/');
-  let result = CONTAINER_MIME_TYPES[container];
-  let transformation = null;
-
-  if (!result) {
-    result = [`video/${container}`, transformation];
-  }
+  const mimetype = CONTAINER_MIME_TYPES[container] || `video/${container}`;
+  let result = [mimetype];
```
[the remark this answers is not in the document]
Comment (tsi) [reply src/plugins/cloudinary/models/video-source/video-source.utils.js:10]: I really prefer avoiding multiple returns where possible

File: src/plugins/cloudinary/models/video-source/video-source.utils.js [hunk 4]
```diff
@@ -6,16 +6,12 @@ import { some } from '../../../../utils/array';
...
 
   if (codec) {
-    transformation = codecToSrcTransformation(codec);
-    result = [`${result[0]}; codecs="${codecShorthandTrans(codec)}"`, transformation];
+    const transformation = codecToSrcTransformation(codec);
+    result = [`${CONTAINER_MIME_TYPES[container]}; codecs="${codecShorthandTrans(codec)}"`, transformation];
```
[the remark this answers is not in the document]
Comment (tsi) [reply src/plugins/cloudinary/models/video-source/video-source.utils.js:14]: That's not clear, formatting was messed up?
Anyway,  `container` should never be undefined, as it iterates over sourceTypes array

File: src/utils/cloudinary.js (hunk 2 above)
Comment (tsi) [reply src/utils/cloudinary.js:37]: I'd like to do that, but we don't have dev-mode yet (I plan to add it) and this is logged only on error and is now much less noisy and helps debugging, so I guess it's ok for now

**Pull request merged**